### PR TITLE
Mention known limitations for --squash

### PR DESCRIPTION
**Pull request by thaJeztah**
The experimental `docker build --squash` feature has a number of limitations. For most use-cases, multi-stage builds are a better alternative.

This patch lists the known limitations, and adds a link to multi-stage builds.

relates to https://github.com/moby/moby/issues/33823

ping @ijc @cpuguy83 @vdemeester

### CODE REVIEW

File: docs/reference/commandline/build.md [hunk 1]
```diff
@@ -473,7 +473,7 @@ FROM alpine AS production-env
 $ docker build -t mybuildimage --target build-env .
 ```
 
-### Squash an image's layers (--squash) **Experimental Only**
+### Squash an image's layers (--squash) (experimental)
```
Comment (thaJeztah) [inline docs/reference/commandline/build.md]: changed this to be in line with other "experimental" features in the docs

File: docs/reference/commandline/build.md [hunk 2]
```diff
@@ -483,13 +483,37 @@ image with the content of the squashed layers. This effectively makes it look
...
+
+The `--squash` option has a number of known limitations:
+
+- When squashing layers, the resulting image cannot take advantage of layer 
+  sharing with other images, and may use significantly more space. Sharing the
+  base image is still supported.
+- When using this option you may see significantly more space used due to
+  storing two copies of the image, one for the build cache with all the cache
+  layers in tact, and one for the squashed version.
+- While squashing layers may produce smaller images, it may
```
Comment (ijc) [inline docs/reference/commandline/build.md]: The end of this sentence is missing.

Comment (thaJeztah) [reply docs/reference/commandline/build.md]: ugh, let me see if it's in my history, haha

Comment (thaJeztah) [reply docs/reference/commandline/build.md]: @ijc updated; I think I had something like that 😅

**Pull request merged**